### ISSUE TITLE
Ctrl-c in Powershell inside Neovim Terminal

### ISSUE THREAD

**Issue by Acelya-9028** [neovim/neovim#13350]
- `nvim --version`:

```
NVIM v0.4.4
Build type: RelWithDebInfo
LuaJIT 2.0.5
Compilation: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe /DWIN32 /D_WINDOWS /W3 /MD /Zi /O2 /Ob1 /DNDEBUG -DMIN_LOG_LEVEL=3 /W3 -D_CRT_SECURE_NO_WARNINGS -D_CRT_NONSTDC_NO_DEPRECATE -DWIN32 -D_WIN32_WINNT=0x0600 -DINCLUDE_GENERATED_DECLARATIONS -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -IC:/projects/neovim/build/config -IC:/projects/neovim/src -IC:/projects/nvim-deps/usr/include -IC:/projects/neovim/build/src/nvim/auto -IC:/projects/neovim/build/include
Compilé par appveyor@APPVYR-WIN

Features: -acl +iconv +tui
See ":help feature-compile"

         fichier vimrc système : "$VIM\sysinit.vim"
               $VIM par défaut : "C:/Program Files/nvim/share/nvim"

Run :checkhealth for more info
```

- `vim -u DEFAULTS` (version: ) behaves differently? No
- Operating system/version:

```
Nom du système d’exploitation:              Microsoft Windows 10 Entreprise LTSC
Version du système:                         10.0.17763 N/A version 17763
```

- Terminal name/version: 

```
echo &term
nvim
```

- `$TERM`:

Default Neovim and Powershell/Powershell Core configs used.

### Actual behaviour

When I press `Ctrl+c`, Powershell Core (pwsh) print `c` instead of call `CopyOrCancelLine`  `PSReadLine`  function.
When I press `Ctrl+c`, Powershell (powershell) print `^C` instead of call `CopyOrCancelLine`  `PSReadLine`  function.

[Powershell Mapping](https://i.imgur.com/3Zrntqc.png)

Try with another keybinding `Set-PSReadLineKeyHandler -Chord Ctrl+n -Function CancelLine` works.
Press `Ctrl-c` with `Set-PSReadLineKeyHandler -Chord c -Function CancelLine` works too, but kill my `c` x).

I tried a `tnoremap <C-c> <Nop>` but press `Ctrl-c` does nothing.
I tried a `tnoremap <C-c> <C-c>` but press `Ctrl-c` print `c` again.

[powershell](https://i.imgur.com/xb8669m.png)
[powershell core](https://i.imgur.com/0Z6Z87E.png)

I have the impression that the Neovim terminal does not send the correct sequence to Powershell but I don't know how to check this.

So my current solution to have a working `Ctrl-c`:

```
tnoremap <C-c> <C-n>
Set-PSReadLineKeyHandler -Chord Ctrl+n -Function CopyOrCancelLine
```

### Expected behaviour

`Ctrl-c` working by default or a remapable `Ctrl-c` to avoid going through an intermediate  key binding.

**Comment by 3N4N** [comment]
TL;DR: seems to have been fixed.  I'm slapping a `needs:response` label.

---

I use PSReadLine but I don't have any experience with `CopyOrCanceline` so what follows might be completely irrelevant.

From what I can gather, `CopyOrCanceline` either copies selected text (if in selection mode) or cancels current input (if in insert mode, that is, if not in select mode).  The thing is, select mode is only available in the old ConHost window, not in Windows Terminal.  Which means you have to use ConHost window to face the issue described in this ticket.

So effectively the __repro steps__ are:
- Open PowerShell through Run
  - Press `Win+R`
  - pwsh
- Launch default nvim: `nvim --clean`
- Ensure you have PSReadLine module for PowerShell installed
- Open powershell inside nvim: `:te pwsh` or `:te powershell`
- Disable mouse support in nvim: `:set mouse=` (see [†] below for reason)
- Go into terminal mode: Press `i` in the nvim `:terminal` window.
- Bind key: `Set-PSReadLineKeyHandler Ctrl+c -Function CopyOrCancelLine`
- Press Enter if you haven't (just being thorough)
- Select some text with mouse: for referencing later, let's select the `Set-PSReadLineKeyHandler` line
- Press `Ctrl+c`
- Check the clipboard content: it should contain the selected text

For me it _is_ copying the selected text to clipboard.  If OP was saying _this_ was the bug, I guess it has been fixed.  If they were facing some other issue, I have misread the whole situation.

† With mouse enabled, you cannot get into ConHost window's Select mode by dragging mouse over text.

**Comment by Acelya-9028** [comment]
This was the bug I think, because it is a 2.5 years old bug and my memory not perfect, I can't be sure and I need to retest. But since I have dropped this thing in the abyss x)
